### PR TITLE
feat: add tinyEngineBaseConfig as base viteconfig

### PR DESCRIPTION
**Pull request by chilingling**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
  - Introduced `htmlUpgradeHttpsPlugin` to enhance security by upgrading HTTP requests to HTTPS.
  - Added `devAliasPlugin` to simplify module referencing during development.

- **Refactors**
  - Updated Vite configuration setup to use `useTinyEngineBaseConfig`, streamlining configuration and improving maintainability.
  - Reorganized configuration logic across multiple files for better code structure.

### CODE REVIEW

File: designer-demo/vite.config.js [hunk 1]
```diff
@@ -1,85 +1,19 @@
 import path from 'node:path'
-import { defineConfig, mergeConfig, loadEnv } from 'vite'
-import { getDefaultConfig } from '@opentiny/tiny-engine-vite-config'
+import { tinyEnginePreset } from '@opentiny/tiny-engine-vite-config'
 
-export default defineConfig((options) => {
+export default tinyEnginePreset({
```
Comment (hexqi) [inline designer-demo/vite.config.js]: 感觉不太符合preset的概念，preset应该是一个插件的集合，不应该是封装defineConfig成为一个高阶函数，并且用户也没法干预merge后的配置，你可以看下[preact-vite的这个preset](https://github.com/preactjs/preset-vite)：
image

File: packages/build/vite-config/index.js [hunk 2]
```diff
@@ -1,2 +1,2 @@
-export { default as getDefaultConfig } from './src/default-config.js'
+export { useTinyEngineBaseConfig } from './src/default-config.js'
```
[the remark this answers is not in the document]
Comment (chilingling) [reply packages/build/vite-config/index.js:1]: already fix,please check again

[the remark this answers is not in the document]
Comment (chilingling) [reply packages/build/vite-config/index.js:1]: the declaration `getDefaultConfig`  inside `packages/build/vite-config/src/default-config.js` is intend to remain and will be call inside the file. just make sure no reference outside of this file is ok.

File: packages/engine-cli/template/designer/vite.config.js [hunk 3]
```diff
@@ -1,34 +1,22 @@
...
-        )
-      : ''
-  }
+export default defineConfig((configEnv) => {
+  const baseConfig = useTinyEngineBaseConfig({
+    viteConfigEnv: configEnv,
+    root: __dirname,
+    iconDirs: [path.resolve(__dirname, './node_modules/@opentiny/tiny-engine/assets/')],
+    envDir: './env'
+  })
```
[the remark this answers is not in the document]
Comment (chilingling) [reply packages/engine-cli/template/designer/vite.config.js:11]: thanks, i will be documenting thoes paramters later.

**Review by hexqi — APPROVED** [review]

**Pull request merged**